### PR TITLE
#126 - Add code coverage support improve

### LINKED ISSUE #126: Improve code-coverage & add GH check action
## Background
Improve or add code coverage support to be able measure code quality.

## Feature
New ability to measure current code coverage as one of QA metrics.
Add GH action to check changed file coverage.

### PR DESCRIPTION
**Pull request by miroslavpojer**
- Added support for code coverage GitHub action check as part of PR.
- Used Absa fork of JaCoCo solution - with scala method filtering.

Closes #126

### CODE REVIEW

File: .github/workflows/check-jacoco.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+#
+# Copyright 2022 ABSA Group Limited
```
Comment (miroslavpojer) [inline .github/workflows/check-jacoco.yml:2]:
```suggestion
# Copyright 2023 ABSA Group Limited
```

**Review by wajda — APPROVED** [review]

**Review by wajda — APPROVED** [review]

File: .github/workflows/check-jacoco.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
+#
+# Copyright 2023 ABSA Group Limited
```
Comment (Zejnilovic) [inline .github/workflows/check-jacoco.yml:2]:
```suggestion
# Copyright 2019 ABSA Group Limited
```

**Comment by cerveada** [comment]
Since Absa JaCoCo 0.8.10 version is out, should it be used?

**Comment by miroslavpojer** [comment]
> Since Absa JaCoCo 0.8.10 version is out, should it be used?

Unfortunately, no. We have to use Absa fork until I will have a capacity to implement logic for origin project and get PR approved. It will take some time as I have defined another priorities for near future.

File: pom.xml [hunk 3]
```diff
@@ -553,9 +553,9 @@
             <build>
                 <plugins>
                     <plugin>
-                        <groupId>org.jacoco</groupId>
+                        <groupId>za.co.absa.jacoco</groupId>
                         <artifactId>jacoco-maven-plugin</artifactId>
-                        <version>0.8.8</version>
+                        <version>0.8.9</version>
```
Comment (cerveada) [inline pom.xml:558]: What I meant was to just use the latest version of the absa fork
```suggestion
                        <version>0.8.10</version>
```

Comment (miroslavpojer) [reply pom.xml]: Sure, I have found not finished push on my local. Solved. Now you should see usage of latest version of Jacoco Absa fork.

**Review by cerveada — APPROVED** [review]

**Review by Zejnilovic — APPROVED** [review]

**Pull request merged**